index.js

```diff
@@ -173,22 +188,28 @@ DB.prototype.del = function (key, opts, cb) {
   })
 }
 
-DB.prototype._del = function (key, opts, cb) {
+// OsmVersion, Opts -> [OsmBatchOp]
+DB.prototype._getDocumentDeletionBatchOps = function (key, opts, cb) {
   var self = this
-  self.kv.get(key, function (err, values) {
+  self.kv.get(key, function (err, docs) {
     if (err) return cb(err)
-    // Filter deletions & map to old expected format.
-    values = filterObj(values, function (key, value) {
-      return !value.deleted
-    })
-    values = mapObj(values, function (key, value) {
-      return value.value
+
+    docs = mapObj(docs, function (version, value) {
+      if (value.deleted) {
+        return {
+          id: key,
+          version: version,
+          deleted: true
+        }
+      } else {
+        return value.value
+      }
     })
 
     var fields = {}
-    var links = opts.keys || Object.keys(values)
+    var links = opts.keys || Object.keys(docs)
     links.forEach(function (ln) {
-      var v = values[ln] || {}
+      var v = docs[ln] || {}
```

Comment:
This is not a bad state issue. We have always been able to create edits of versions that are not the current head. `put` accepts `opts.links` and `del` should have. If you edit/delete a version that is not the head you just create a fork, including with a deletion. `opts.keys` is a typo, I thought I had opened an issue for that but I can't see it. This is a bug which has existed for a while in this deletion code I realize.